src/SFML/Graphics/Shape.cpp

```diff
@@ -143,6 +145,38 @@ FloatRect Shape::getGlobalBounds() const
     return getTransform().transformRect(getLocalBounds());
 }
 
+Vector2f Shape::getGeometricCenter() const
+{
+    const auto count = getPointCount();
+
+    switch (count)
+    {
+        case 0:
+            err() << "Shape::getGeometricCenter() called on empty Shape";
+            std::terminate();
+        case 1:
+            return getPoint(0);
+        case 2:
+            return (getPoint(0) + getPoint(1)) / 2.f;
+        default: // more than two points
+            Vector2f centroid;
+            float    twicearea = 0;
+
+            auto p2 = getPoint(count - 1);
```

Comment:
Is there any way to give this a more descriptive name? I'm not sure how to interpret `p1` and `p2`. Is `p2` supposed to be the previous point as you go through all points in the shape?

Reply:
Yepp, that's it. Renamed them now.